src/test/java/org/cactoos/http/HtTimedWireTest.java

```diff
@@ -76,24 +84,131 @@ public void worksFine() throws Exception {
         );
     }
 
-    // @todo #78:30min With #78 solved, HtWire stopped closing the connection
-    //  when send() is called and the following test stopped working for no
-    //  clear reason. Investigate the cause of this, fix it and unignore this.
-    @Ignore("see todo above")
     // @checkstyle MagicNumberCheck (1 line)
-    @Test(expected = TimeoutException.class, timeout = 1000)
+    @Test(timeout = 1000)
     public void failsAfterTimeout() throws Exception {
         // @checkstyle MagicNumberCheck (1 line)
         final long timeout = 100;
-        try (Socket blocker = new Socket()) {
-            blocker.connect(this.server.getLocalSocketAddress());
-            new HtTimedWire(
-                new HtWire(
-                    this.server.getInetAddress().getHostAddress(),
-                    this.server.getLocalPort()
-                ),
-                timeout
-            ).send(new InputOf("unused"));
+        final Wire wire = new HtTimedWire(
+            input -> {
+                TimeUnit.SECONDS.sleep(timeout + 1);
+                return input;
+            },
+            timeout
+        );
+        new Assertion<>(
+            "fails with timeout exception",
+            () -> wire.send(new InputOf()),
+            new TimeoutExceptionMatcher()
+        ).affirm();
+    }
+
+    // @checkstyle MagicNumberCheck (1 line)
+    @Test(timeout = 1000)
+    @SuppressWarnings("unchecked")
+    // @todo #87:30m In the creation of IterableOf without
+    //  @SuppressWarnings("unchecked") complication failed with:
+    //  Warning:(124, 29) java: unchecked generic array creation
+    //  for varargs parameter of type
+    //  org.hamcrest.Matcher<? super org.cactoos.Scalar<org.cactoos.Input>>[].
+    //  Needs to investigate and fix this warning.
+    public void failsAfterTimeoutCheckIfWait() throws Exception {
+        final long timeout = 100;
+        final long sleep = 10 * timeout;
+        final Wire wire = new HtTimedWire(
+            input -> {
+                TimeUnit.SECONDS.sleep(sleep);
+                return input;
+            },
+            timeout
+        );
+        final long start = System.currentTimeMillis();
+        final JoinedText message = new JoinedText(
```

Comment:
@aivinog1 the assertion's message should express a single assertion. Asserting two things means you're testing two things at the same time

Reply:
I agree.